### PR TITLE
Update plugin to obtain the docker tag from gradle.properties file

### PR DESCRIPTION
**Pull request by MohamedSabthar**
Fixes: https://github.com/ballerina-platform/ballerina-library/issues/5859

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: src/main/groovy/io/ballerina/plugin/BallerinaPlugin.groovy [hunk 1]
```diff
@@ -79,9 +79,9 @@ class BallerinaPlugin implements Plugin<Project> {
         }
 
         project.afterEvaluate {
-            if (ballerinaExtension.buildOnDockerImage != null) {
+            if (ballerinaExtension.hasProperty('buildOnDockerImage') && ballerinaExtension.buildOnDockerImage.asType(boolean) == true) {
```
Comment (ThisaruGuruge) [inline src/main/groovy/io/ballerina/plugin/BallerinaPlugin.groovy:82]: Shall we check whether the parameter is an actual boolean, before we process it?

File: src/main/groovy/io/ballerina/plugin/BallerinaPlugin.groovy [hunk 2]
```diff
@@ -510,4 +510,8 @@ class BallerinaPlugin implements Plugin<Project> {
             println("Failed to delete $filePath.")
         }
     }
+
+    static String extractVersion(String versionWithTimeStamp) {
+        return versionWithTimeStamp.trim().split("-")[0]
+    }
```
Comment (niveathika) [inline src/main/groovy/io/ballerina/plugin/BallerinaPlugin.groovy:516]: Will this work for un released versions? For instance if we depend on update 9 lang timestamped version. The docker tag will not be available. Lets go with nightly in that case.

File: src/main/groovy/io/ballerina/plugin/BallerinaPlugin.groovy [hunk 3]
```diff
@@ -325,7 +325,7 @@ class BallerinaPlugin implements Plugin<Project> {
                             workingDir project.projectDir
                             environment 'JAVA_OPTS', '-DBALLERINA_DEV_COMPILE_BALLERINA_ORG=true'
                             if (buildOnDocker) {
-                                String dockerTag = ballerinaExtension.buildOnDockerImage
+                                String dockerTag = extractVersion(project.findProperty('ballerinaLangVersion'))
```
Comment (ThisaruGuruge) [inline src/main/groovy/io/ballerina/plugin/BallerinaPlugin.groovy:328]: We should have an option to override the docker tag using a Gradle CLI argument. (We have that functionality right now, just have to update the logic accordingly). 

So, the process should be,

- Check whether this is a connector build (Maybe we can change the input parameter to reflect the question, "is this a connector build or not"
- If this is a connector build, get the docker tag using the `ballerinaLangVersion` property from the Gralde properties file
- But if the user has provided the `-P buildUsingDocker=<dockerTag>` CLI argument, override the previously extracted docker version using the CLI argument.

Comment (MohamedSabthar) [reply src/main/groovy/io/ballerina/plugin/BallerinaPlugin.groovy]: Updated with [a6da00c](https://github.com/ballerina-platform/plugin-gradle/pull/122/commits/a6da00cdc497ede8ac995a0e2c7c8a18a91ce097)

File: src/main/groovy/io/ballerina/plugin/BallerinaPlugin.groovy (hunk 1 above)
Comment (MohamedSabthar) [reply src/main/groovy/io/ballerina/plugin/BallerinaPlugin.groovy]: parameter changed to boolean type now

File: src/main/groovy/io/ballerina/plugin/BallerinaPlugin.groovy (hunk 2 above)
Comment (MohamedSabthar) [reply src/main/groovy/io/ballerina/plugin/BallerinaPlugin.groovy]: Updated with [a6da00c](https://github.com/ballerina-platform/plugin-gradle/pull/122/commits/a6da00cdc497ede8ac995a0e2c7c8a18a91ce097)

**Review by ThisaruGuruge — APPROVED** [review]

**Pull request merged**